sqlglot/generator.py

```diff
@@ -3648,24 +3654,42 @@ def gt_sql(self, expression: exp.GT) -> str:
     def gte_sql(self, expression: exp.GTE) -> str:
         return self.binary(expression, ">=")
 
-    def ilike_sql(self, expression: exp.ILike) -> str:
-        return self.binary(expression, "ILIKE")
-
-    def ilikeany_sql(self, expression: exp.ILikeAny) -> str:
-        return self.binary(expression, "ILIKE ANY")
-
     def is_sql(self, expression: exp.Is) -> str:
         if not self.IS_BOOL_ALLOWED and isinstance(expression.expression, exp.Boolean):
             return self.sql(
                 expression.this if expression.expression.this else exp.not_(expression.this)
             )
         return self.binary(expression, "IS")
 
+    def _like_sql(self, expression: exp.Like | exp.ILike) -> str:
+        this = expression.this
+        rhs = expression.expression
+
+        if isinstance(expression, exp.Like):
+            exp_class: t.Type[exp.Like | exp.ILike] = exp.Like
+            op = "LIKE"
+        else:
+            exp_class = exp.ILike
+            op = "ILIKE"
+
+        if isinstance(rhs, (exp.All, exp.Any)) and not self.SUPPORTS_LIKE_QUANTIFIERS:
+            exprs = [rhs.this.this] if isinstance(rhs.this, exp.Paren) else rhs.this.expressions
+            func = exp.or_ if isinstance(rhs, exp.Any) else exp.and_
+
+            like_sql: exp.Expression = exp_class(this=this, expression=exprs[0])
+
+            for expr in exprs[1:]:
+                like_sql = func(like_sql, exp_class(this=this, expression=expr))
```

Comment:
We can also `reduce` here:
```suggestion
            like_expr = reduce(lambda x, y: func(exp_class(this=x, expression=y)), [this, *exprs])
```